src/bip39.cpp

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright (c) 2013-2014 Tomas Dzetkulic
+ * Copyright (c) 2013-2014 Pavol Rusnak
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES
+ * OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+// Source:
+// https://github.com/trezor/trezor-crypto
+
+#include <string.h>
+#include <stdbool.h>
+
+#include "bip39.h"
+// #include "hmac.h"
+// #include "rand.h"
+// #include "sha2.h"
+// #include "pbkdf2.h"
+#include "bip39_english.h"
+// #include "options.h"
+#include "crypto/sha256.h"
+#include "random.h"
+
+#include <openssl/evp.h>
+
+const char *mnemonic_generate(int strength)
+{
+    if (strength % 32 || strength < 128 || strength > 256) {
+        return 0;
+    }
+    uint8_t data[32];
+    // random_buffer(data, 32);
+    GetRandBytes(data, 32);
+    return mnemonic_from_data(data, strength / 8);
+}
+
+const char *mnemonic_from_data(const uint8_t *data, int len)
```

Comment:
Make the return type `std::string` and remove  the static local  `static char mnemo[24 * 10];`
The only caller, `hdchain.cpp` line 86 converts to (i.e. copies into) a `std::string` anyway.